packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1130,8 +1169,29 @@ class RenderEditable extends RenderBox {
   double get preferredLineHeight => _textPainter.preferredLineHeight;
 
   double _preferredHeight(double width) {
-    if (maxLines != null)
+    // Lock height to maxLines if needed
+    final bool lockedMax = maxLines != null && minLines == null;
+    final bool lockedBoth = minLines != null && minLines == maxLines;
```

Comment:
I think the logic is the same, so:

    final bool lockedBoth = minLines != null && minLines == maxLines;
    // is the same as:
    final bool lockedBoth = minLines != null && maxLines !== null && minLines == maxLines;

I want to exclude cases like `minLines: 2, maxLines: 4`, where it will grow and not be locked.

Let me know if I'm misunderstanding your comment.